builtin/credential/approle/path_role.go

```diff
@@ -840,14 +875,28 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
 		role.BindSecretID = data.Get("bind_secret_id").(bool)
 	}
 
-	if boundCIDRListRaw, ok := data.GetOk("bound_cidr_list"); ok {
-		role.BoundCIDRList = boundCIDRListRaw.([]string)
-	} else if req.Operation == logical.CreateOperation {
-		role.BoundCIDRList = data.Get("bound_cidr_list").([]string)
+	if boundCIDRListRaw, ok := data.GetOk("secret_id_bound_cidrs"); ok {
+		role.SecretIDBoundCIDRs = boundCIDRListRaw.([]string)
+	} else if boundCIDRListRaw, ok := data.GetOk("bound_cidr_list"); ok {
+		role.SecretIDBoundCIDRs = boundCIDRListRaw.([]string)
 	}
 
-	if len(role.BoundCIDRList) != 0 {
-		valid, err := cidrutil.ValidateCIDRListSlice(role.BoundCIDRList)
+	if len(role.SecretIDBoundCIDRs) != 0 {
+		valid, err := cidrutil.ValidateCIDRListSlice(role.SecretIDBoundCIDRs)
+		if err != nil {
+			return nil, errwrap.Wrapf("failed to validate CIDR blocks: {{err}}", err)
+		}
+		if !valid {
+			return logical.ErrorResponse("invalid CIDR blocks"), nil
+		}
+	}
+
+	if boundCIDRListRaw, ok := data.GetOk("token_bound_cidrs"); ok {
```

Comment:
These three lines can just be `role.TokenBoundCIDRs = data.Get("token_bound_cidrs").([]string)`

Reply:
I thought that initially too, but it turned out that if I did that, then on the update operation it wiped out any previously set values if that field wasn't explicitly set by the caller. So I added the ok check so that I'd only overwrite the value if it was actually sent to me.

Reply:
Oh, right, I forgot it's create/update, not just update! Sounds good.